### PR TITLE
Replacing hand rolled rate limiter with Gauva's Rate Limiter.

### PR DESCRIPTION
**Pull request by domesticmouse**
PTAL @markmcd 

This is a fix for #261

### CODE REVIEW

File: src/main/java/com/google/maps/OkHttpRequestHandler.java [hunk 1]
```diff
@@ -102,11 +102,9 @@ public void setQueriesPerSecond(int maxQps) {
     rateLimitExecutorService.setQueriesPerSecond(maxQps);
   }
 
-  @Override
+  @Override @Deprecated
   public void setQueriesPerSecond(int maxQps, int minimumInterval) {
-    dispatcher.setMaxRequests(maxQps);
-    dispatcher.setMaxRequestsPerHost(maxQps);
-    rateLimitExecutorService.setQueriesPerSecond(maxQps, minimumInterval);
+    this.setQueriesPerSecond(maxQps);
```
Comment (markmcd) [inline src/main/java/com/google/maps/OkHttpRequestHandler.java]: Can you log a warning or something here? I'm concerned that ignoring an explicitly provided parameter will lead to unexpected behaviour for users.

File: src/test/java/com/google/maps/model/EnumsTest.java [hunk 2]
```diff
@@ -147,6 +147,7 @@ public void testCanonicalLiteralsForAddressComponentType() {
     addressComponentTypeToLiteralMap.put(AddressComponentType.TRAIN_STATION, "train_station");
     addressComponentTypeToLiteralMap.put(AddressComponentType.SUBWAY_STATION, "subway_station");
     addressComponentTypeToLiteralMap.put(AddressComponentType.TRANSIT_STATION, "transit_station");
+    addressComponentTypeToLiteralMap.put(AddressComponentType.LIGHT_RAIL_STATION, "light_rail_station");
```
Comment (markmcd) [inline src/test/java/com/google/maps/model/EnumsTest.java]: This appears to be unrelated to your proposed change.

Comment (domesticmouse) [inline src/test/java/com/google/maps/model/EnumsTest.java]: I can put it into a separate PR, it's a fix for a test failure.

File: src/main/java/com/google/maps/OkHttpRequestHandler.java (hunk 1 above)
Comment (domesticmouse) [inline src/main/java/com/google/maps/OkHttpRequestHandler.java]: Done
